primeqa/qg/processors/passage_qg/tydiqa_processor.py

```diff
@@ -0,0 +1,18 @@
+from tqdm import tqdm
+from datasets import load_dataset
+from primeqa.qg.utils.constants import QGSpecialTokens
+
+class TydiQADataset():
+	def __init__(self, dataset_name='tydiqa'):
+		self.dataset_name = dataset_name
+
+	def preprocess_data_for_qg(self, data_split='train'):
+		data = load_dataset(self.dataset_name, 'secondary_task', split=data_split)
+		processed_data_dict = {'question':[], 'input':[]}
+		
+		for d in tqdm(data):
+			input = d['answers']['text'][0] +' '+QGSpecialTokens.sep+' ' + d['context']
```

Comment:
Is this coming from HF? This doesn't look like the TyDi format.

Reply:
This is coming from HF, tydi secondary task. This preprocessor code aim towards making it suitable for QG training, that's why it might look a little different than the mrc. Test cases are added to make sure the format is correct.